common/src/main/java/coffee/waffle/emcutils/mixin/PlayerEntityMixin.java

```diff
@@ -0,0 +1,26 @@
+package coffee.waffle.emcutils.mixin;
+
+import coffee.waffle.emcutils.Caches;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.text.Text;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.concurrent.ExecutionException;
+
+@Mixin(PlayerEntity.class)
+public abstract class PlayerEntityMixin {
+
+	@Inject(method = "getDisplayName", at = @At("HEAD"), cancellable = true)
+	private void emcutils$getDisplayName(CallbackInfoReturnable<Text> cir) {
+		PlayerEntity e = ((PlayerEntity) (Object) this);
+
+		try {
+			cir.setReturnValue(Caches.namePlateCache.get(e));
```

Comment:
This variable can just be inlined, unless you think that's less readable

Reply:
I thought that at first, but then there would just be too much happening on that line